sticker/src/tensorflow/lr.rs

```diff
@@ -66,23 +82,38 @@ impl ExponentialDecay {
         );
 
         ExponentialDecay {
+            lr: initial_lr,
             initial_lr,
             decay_rate,
             decay_steps,
             staircase,
+            step: 0,
+            warmup_steps,
         }
     }
 }
 
 impl LearningRateSchedule for ExponentialDecay {
-    fn learning_rate(&mut self, epoch: usize, _last_score: f32) -> f32 {
+    fn compute_batch_learning_rate(&mut self) -> f32 {
+        let lr = if self.step < self.warmup_steps {
+            self.initial_lr / (self.warmup_steps as f32) * self.step as f32
+        } else {
+            self.lr
+        };
+        self.step += 1;
+        lr
+    }
+
+    fn compute_epoch_learning_rate(&mut self, epoch: usize, _last_score: f32) -> f32 {
+        self.step += 1;
```

Comment:
Should `step` be increased here as well, since it is already increased for batches?

Reply:
Thanks, that shouldn't be there.